terraform/production/service_account_roles.tf

```diff
@@ -0,0 +1,200 @@
+resource "aws_iam_role" "full_crud_access_spack_binaries_prs" {
+  name        = "FullCRUDAccessToBucketSpackBinariesPRs"
+  description = "Managed by Terraform. Grants Kubernetes pods access to read/write/delete objects from the spack-binaries-prs S3 bucket."
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Federated" : module.production_cluster.oidc_provider_arn
+        },
+        "Action" : "sts:AssumeRoleWithWebIdentity",
+        "Condition" : {
+          "StringEquals" : {
+            "${module.production_cluster.oidc_provider}:aud" : "sts.amazonaws.com"
+          }
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "put_spack_binaries_prs" {
+  name = "PutObjectsInBucketSpackBinariesPRs"
+  role = aws_iam_role.full_crud_access_spack_binaries_prs.id
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Action" : "s3:PutObject",
+        "Resource" : "arn:aws:s3:::spack-binaries-prs/*"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "delete_spack_binaries_prs" {
+  name = "DeleteObjectsFromBucketSpackBinariesPRs"
+  role = aws_iam_role.full_crud_access_spack_binaries_prs.id
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Action" : "s3:DeleteObject",
+        "Resource" : "arn:aws:s3:::spack-binaries-prs/*"
+      }
+    ]
+  })
+}
+
+resource "kubectl_manifest" "spackbot_service_account" {
+  for_each  = toset(["spackbot", "spackbotdev"])
+  yaml_body = <<-YAML
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: ${each.value}-spack-io
+      namespace: spack
+      annotations:
+        eks.amazonaws.com/role-arn: ${aws_iam_role.full_crud_access_spack_binaries_prs.arn}
+  YAML
+  depends_on = [
+    aws_iam_role_policy.put_spack_binaries_prs,
+    aws_iam_role_policy.delete_spack_binaries_prs
+  ]
+}
+
+resource "aws_iam_role" "put_object_in_pipeline_statistics" {
+  name        = "PutObjectInPipelineStatistics"
+  description = "Managed by Terraform. Grant access to write to the pipeline-statistics folder of the spack-logs bucket."
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Federated" : module.production_cluster.oidc_provider_arn
+        },
+        "Action" : "sts:AssumeRoleWithWebIdentity",
+        "Condition" : {
+          "StringEquals" : {
+            "${module.production_cluster.oidc_provider}:aud" : "sts.amazonaws.com"
+          }
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "put_object_in_pipeline_statistics" {
+  name = "PutObjectInPipelineStatistics"
+  role = aws_iam_role.full_crud_access_spack_binaries_prs.id
```

Comment:
```suggestion
  role = aws_iam_role.put_object_in_pipeline_statistics.id
```

Reply:
Oh wow :sweat: , thanks for catching that!